versioned/tests/src/main/java/org/projectnessie/versioned/tests/AbstractCommitLog.java

```diff
@@ -114,23 +113,26 @@ public void commitLogExtendedNoGlobalState() throws Exception {
 
     int numCommits = 10;
 
-    List<Hash> hashes =
-        IntStream.rangeClosed(1, numCommits)
-            .mapToObj(
-                i -> {
-                  try {
-                    return commit("Commit #" + i)
-                        .put("k" + i, onRef("v" + i, "c" + i))
-                        .put("key" + i, onRef("value" + i, "cid" + i))
-                        .delete("delete" + i)
-                        .toBranch(branch);
-                  } catch (Exception e) {
-                    throw new RuntimeException(e);
-                  }
-                })
-            .collect(Collectors.toList());
+    CommitBuilder init = commit("initial");
+    List<Hash> hashes = new ArrayList<>();
+
+    for (int i = 1; i <= numCommits; i++) {
+      init = init.put("delete" + i, onRef("to delete " + i, "d" + i));
+    }
+    hashes.add(init.toBranch(branch));
+
+    for (int i = 1; i <= numCommits; i++) {
+      Hash head =
+          commit("Commit #" + i)
+              .put("k" + i, onRef("v" + i, "c" + i))
+              .put("key" + i, onRef("value" + i, "cid" + i))
+              .delete("delete" + i)
+              .toBranch(branch);
+      hashes.add(head);
+    }
+
     List<Hash> parentHashes =
-        Stream.concat(Stream.of(firstParent), hashes.subList(0, 9).stream())
+        Stream.concat(Stream.of(firstParent), hashes.subList(0, numCommits + 1).stream())
```

Comment:
`9 == numCommits - 1` :thinking: was is wrong before?

Reply:
It's "shifted by 1" below now, and one more hash added above.

Reply:
ah, I see... I was thrown off balance by "minor refactoring" ;)

Reply:
Do we need `subList` now?